java/arcs/android/devtools/DevWebSocketFactory.kt

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.android.devtools
+
+import arcs.core.util.TaggedLog
+import fi.iki.elonen.NanoHTTPD
+import fi.iki.elonen.NanoWSD
+import fi.iki.elonen.NanoWSD.WebSocketFrame.CloseCode
+import java.io.IOException
+
+/**
+ * An extension of [NanoWSD] for devtools to connect Arcs to a remote device for debugging.
+ */
+class DevWebSocketFactory {
```

Comment:
Typically the Factory pattern when used like this is for an AbstractFactory (something where you can sub-out the factory method/constructor). For this case, I think if we only ever need one webserver, using a simple singleton would suffice.  If we needed to extract an interface for an abstract factory later, it could still be possible.